src/main/scala/kamon/http4s/middleware/client/KamonSupport.scala

```diff
@@ -40,11 +40,10 @@ object KamonSupport {
 
 
   def apply[F[_]](underlying: Client[F])(implicit F:Sync[F]): Client[F] = Client { request =>
+    // this needs to run on the same thread as the caller, so can't be suspended in F
+    val ctx = Kamon.currentContext()
```

Comment:
I think we _could_ if we had a `make` function that returned `F[Client[F]]`?

Nice comment, by the way.

Reply:
How would that work? Not sure it would be worth doing, anyway, given the extra complexity for the caller.

Reply:
True, it's debatable if it's worth or not. We use that pattern in some of our services (e.g. Storm), and we tend to like it.

I was thinking that, if we _wanted_, one option could be:

```
def make[F[_]](underlying: Client[F])(implicit F:Sync[F]): F[Client[F]] = F.delay(Kamon.currentContext()).map { ctx =>
    Client { request => kamonClient(underlying)(request)(ctx)(_instrumentation) }
  }
```

But when writing that, I started to wonder if `Kamon.currentContext()` would then be invoked in the caller thread, and the answer might be 'no'.

Reply:
I'm seeing the code as it is here intermittently fail tests - also, it's not correct, as it uses the context from the thread where the request is instantiated rather than from the thread where it's executed. I think doing this correctly would require writing bytecode instrumentation for the CE3 runtime. That said, I think it's fine for what we're using it for in Personalization since we're just using Kamon for metrics rather than tracing.